aks/application/resources.tf

```diff
@@ -0,0 +1,145 @@
+locals {
+  app_name = "${var.name}-${var.app_environment}"
+}
+
+resource "kubernetes_deployment" "main" {
+  metadata {
+    name      = local.app_name
+    namespace = var.namespace
+  }
+
+  spec {
+    replicas = var.replicas
+
+    selector {
+      match_labels = {
+        app = local.app_name
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = local.app_name
+        }
+      }
+
+      spec {
+        node_selector = {
+          "teacherservices.cloud/node_pool" = "applications"
+          "kubernetes.io/os"                = "linux"
+        }
+
+        container {
+          name    = local.app_name
+          image   = var.docker_image
+          command = try(slice(var.command, 0, 1), null)
+          args    = try(slice(var.command, 1, length(var.command)), null)
+
+          liveness_probe {
```

Comment:
it's very useful to have probes optional e.g. a dynamic block as per https://github.com/DFE-Digital/register-trainee-teachers/blob/main/terraform/modules/kubernetes/main.tf#L44

Reply:
Ah nice, yes `dynamic` is exactly the sort of thing I was looking for to implement this, so thanks!